src/util/charbuf.c

```diff
@@ -13,20 +13,38 @@ charbuf new_charbuf(size_t len)
 {
   charbuf newBuf;
 
-  newBuf.chars = (unsigned char *) malloc(len);
-  newBuf.len = len;
+  newBuf.len = 0;
+  if (len > 0)
+  {
+    newBuf.chars = (unsigned char *) malloc(len);
+    if (newBuf.chars != NULL)
+    {
+      newBuf.len = len;
+    }
+  }
+  else
+  {
+    newBuf.chars = NULL;
+  }
   return (newBuf);
 }
 
 void free_charbuf(charbuf * buf)
 {
-  free((*buf).chars);
-  (*buf).chars = NULL;
-  (*buf).len = 0;
+  if (buf->chars != NULL && buf->len != 0)
```

Comment:
We don't need the NULL check. Freeing NULL is a no-op. This is worth removing because if the pointer is NULL but the length is wrong, this will clean that up, too.

Reply:
Fixed with b25ad49, which also checks that the pointer ```free_charbuf``` is handed isn't ```NULL```.